### ISSUE TITLE
SSR rendering of internal pages does not work on Live

### ISSUE THREAD

**Issue by Sharique-Hasan** [MarkPieszak/aspnetcore-angular2-universal#556]
Hello, I have integrated this structure successfully. It has been deployed on live too. The issue I am facing is that on live the server side rendering of internal pages are not working. It rather renders home page and then route to the requested page. Which is not the expected behavior. However SSR of internal pages is working fine on localhost. I have deployed the application using IIS on live. What am I missing there? I am out of ideas to make it work fine. localhost is working fine where as on live it is not working as expected.

**Comment by MarkPieszak** [comment]
Does it error out when you do `npm run build:prod` locally and try to create all the bundles?

**Comment by Sharique-Hasan** [comment]
I create build from visual studio publish area and it doesn't error out any
thing while creating build.

**Comment by Sharique-Hasan** [comment]
When I try to view the page source of any internal/dynamic page it always give me the home page's page source

**Comment by odahcam** [comment]
Did you try to directly access that page and than see the source?

**Comment by Sharique-Hasan** [comment]
Yes I did that. For eg: myapp.com/merchant-1 first lands on myapp.com and then navigate to myapp.com/merchant1 whEn I tried to view the page source of merchant1 page ot gave me the source of myapp.com

**Comment by odahcam** [comment]
It should work that way, unless you directly access the route of that page.

**Comment by Sharique-Hasan** [comment]
I have to access the page directly in order to have proper seo. There are
dynamic meta descriptions and titles. This is working on localhost but not
working via iis

On Feb 8, 2018 6:15 PM, "Luiz Machado" <notifications@github.com> wrote:

It should work that way, unless you directly access the route of that page.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/MarkPieszak/aspnetcore-angular2-universal/issues/556#issuecomment-364108088>,
or mute the thread

**Comment by Sharique-Hasan** [comment]
Try to view this web page. http://35.158.211.140 and then access the
categories from sidebar. It will route eventually as expected. But when you
view that category page directly it will first land on home page and then
navigate to the reqyested page.

Similarly when you click Avail now button it will take you to detail page.
But directly accessing that page will render the home page first and then
navigate to detail page on client side.

**Comment by odahcam** [comment]
Sorry, but that link opens a blank page for me.
